File: src/minecraft/net/machinemuse/powersuits/network/packets/MusePacketPlasmaBolt.java
package net.machinemuse.powersuits.network.packets;

import java.io.DataInputStream;

import net.machinemuse.powersuits.entity.EntityPlasmaBolt;
import net.machinemuse.powersuits.network.MusePacket;
import net.minecraft.client.Minecraft;
import net.minecraft.client.entity.EntityClientPlayerMP;
import net.minecraft.entity.player.EntityPlayerMP;
import cpw.mods.fml.common.network.Player;

public class MusePacketPlasmaBolt extends MusePacket {
	protected int entityID;
	protected double size;

	public MusePacketPlasmaBolt(Player player, int entityID, double size) {
		super(player);
		writeInt(entityID);
		writeDouble(size);

	}

	public MusePacketPlasmaBolt(DataInputStream data, Player player) {
		super(data, player);
		entityID = readInt();
		size = readDouble();
	}

	@Override
	public void handleClient(EntityClientPlayerMP player) {
		try {
			EntityPlasmaBolt entity = (EntityPlasmaBolt) Minecraft.getMinecraft().theWorld.getEntityByID(entityID);
			entity.size = this.size;
		} catch (Exception e) {
			return;
		}

	}

	@Override
	public void handleServer(EntityPlayerMP player) {
		// TODO Auto-generated method stub

	}

}

File: src/minecraft/net/machinemuse/powersuits/network/packets/MusePacketPlasmaBolt.scala
package net.machinemuse.powersuits.network.packets

import java.io.DataInputStream
import net.machinemuse.powersuits.entity.EntityPlasmaBolt
import net.machinemuse.powersuits.network.{MusePackager, MusePacket}
import net.minecraft.client.Minecraft
import net.minecraft.client.entity.EntityClientPlayerMP
import cpw.mods.fml.common.network.Player

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 12:28 PM, 5/6/13
 */
object MusePacketPlasmaBolt extends MusePackager {
  def read(d: DataInputStream, p: Player) = {
    val entityID = readInt(d)
    val size = readDouble(d)
    new MusePacketPlasmaBolt(p, entityID, size)
  }
}

class MusePacketPlasmaBolt(player: Player, entityID: Int, size: Double) extends MusePacket(player) {
  val packager = MusePacketPlasmaBolt

  def write {
    writeInt(entityID)
    writeDouble(size)
  }

  override def handleClient(player: EntityClientPlayerMP) {
    try {
      val entity: EntityPlasmaBolt = Minecraft.getMinecraft.theWorld.getEntityByID(entityID).asInstanceOf[EntityPlasmaBolt]
      entity.size = this.size
    }
    catch {
      case e: Exception => {
        return
      }
    }
  }
}